# 02 - Experimentos Randomizados

## O Padrão Ouro

Na sessão anterior, vimos por que e como a associação difere da causalidade. Também vimos o que é necessário para que uma associação se torne causalidade.

$
E[Y|T=1] - E[Y|T=0] = \underbrace{E[Y_1 - Y_0|T=1]}_{ATT} + \underbrace{\{ E[Y_0|T=1] - E[Y_0|T=0] \}}_{BIAS}
$

Para recapitular, a associação se torna causalidade se não houver viés. Não haverá viés se \(E[Y_0|T=0] = E[Y_0|T=1]\). Em outras palavras, a associação será causalidade se os grupos tratado e controle forem iguais ou comparáveis, exceto pelo tratamento recebido. Ou, em termos mais técnicos, quando o resultado do não tratado é igual ao resultado contrafactual do tratado. Lembre-se de que este resultado contrafactual é o resultado do grupo tratado se eles não tivessem recebido o tratamento.

Acho que fizemos um trabalho razoável explicando como fazer a associação ser igual à causalidade em termos matemáticos. Mas isso foi apenas em teoria. Agora, vamos olhar para a primeira ferramenta que temos para fazer o viés desaparecer: **Experimentos Randomizados**. Experimentos randomizados atribuem aleatoriamente indivíduos em uma população a um grupo de tratamento ou a um grupo controle. A proporção que recebe o tratamento não precisa ser de 50%. Você poderia ter um experimento onde apenas 10% das suas amostras recebem o tratamento.

A randomização aniquila o viés ao tornar os resultados potenciais independentes do tratamento.

$
(Y_0, Y_1) \perp\!\!\!\perp T
$

Isso pode ser confuso no início (foi para mim). Mas não se preocupe, meu corajoso e verdadeiro companheiro, eu explicarei mais detalhadamente. Se o resultado é independente do tratamento, isso não implica também que o tratamento não tem efeito? Bem, sim! Mas observe que não estou falando sobre os resultados. Em vez disso, estou falando sobre os **resultados potenciais**. O resultado potencial é como o resultado **teria sido** sob tratamento (\(Y_1\)) ou sob controle (\(Y_0\)). Em ensaios randomizados, **não** queremos que o resultado seja **independente** do tratamento, já que pensamos que **o tratamento causa** o resultado. Mas queremos que os **resultados potenciais** sejam independentes do tratamento.

![img](./data/img/rct/indep.png)

Dizer que os resultados potenciais são independentes do tratamento significa que eles seriam, em expectativa, os mesmos tanto no grupo de tratamento quanto no grupo de controle. Em termos mais simples, isso significa que os grupos de tratamento e controle são comparáveis. Ou que saber a atribuição do tratamento não me dá nenhuma informação sobre como era o resultado antes do tratamento. Consequentemente, \((Y_0, Y_1) \perp T\) significa que o tratamento é a única coisa gerando uma diferença entre o resultado no grupo tratado e no grupo de controle. Para ver isso, observe que a independência implica precisamente que

$
E[Y_0|T=0]=E[Y_0|T=1]=E[Y_0]
$

O que, como vimos, faz com que

$
E[Y|T=1] - E[Y|T=0] = E[Y_1 - Y_0]=ATE
$

Portanto, a randomização nos oferece uma maneira de usar uma simples diferença nas médias entre tratamento e controle e chamar isso de efeito do tratamento.


## Em uma Escola Muito, Muito Distante

Em 2020, a pandemia do Coronavírus forçou as empresas a se adaptarem ao distanciamento social. Os serviços de entrega se tornaram generalizados e grandes corporações adotaram estratégias de trabalho remoto. Com as escolas, não foi diferente. Muitas iniciaram seus próprios repositórios online de aulas.

Quatro meses após o início da crise, muitos se perguntam se as mudanças introduzidas poderiam ser mantidas. Não há dúvida de que o aprendizado online tem seus benefícios. É mais barato, pois pode economizar em imóveis e transporte. Também pode ser mais digital, aproveitando conteúdo de classe mundial de todo o globo, não apenas de um conjunto fixo de professores. Apesar de tudo isso, ainda precisamos responder se o aprendizado online tem um impacto negativo ou positivo no desempenho acadêmico dos alunos.

Uma maneira de responder a isso é pegar estudantes de escolas que oferecem principalmente aulas online e compará-los com estudantes de escolas que oferecem aulas em salas de aula tradicionais. Como sabemos agora, essa não é a melhor abordagem. Pode ser que as escolas online atraiam apenas os alunos bem disciplinados que se saem melhor do que a média mesmo que a aula fosse presencial. Neste caso, teríamos um viés positivo, onde os tratados são academicamente melhores do que os não tratados: \(E[Y_0|T=1] > E[Y_0|T=0]\).

Ou, por outro lado, pode ser que as aulas online sejam mais baratas e sejam compostas principalmente por alunos menos abastados, que podem ter que trabalhar além de estudar. Neste caso, esses alunos se sairiam pior do que aqueles das escolas presenciais, mesmo se eles participassem de aulas presenciais. Se este fosse o caso, teríamos um viés na outra direção, onde os tratados são academicamente piores do que os não tratados: \(E[Y_0|T=1] < E[Y_0|T=0]\).

Portanto, embora pudéssemos fazer comparações simples, isso não seria convincente. De uma forma ou de outra, nunca poderíamos ter certeza de que não haveria nenhum viés por perto mascarando nosso efeito causal.

![img](./data/img/rct/lurking_bias.png)

Para resolver isso, precisamos tornar os tratados e não tratados comparáveis \(E[Y_0|T=1] = E[Y_0|T=0]\). Uma maneira de forçar isso é atribuindo aleatoriamente as aulas online e presenciais aos estudantes. Se conseguíssemos fazer isso, os tratados e não tratados seriam, em média, os mesmos, exceto pelo tratamento que recebem.

Felizmente, alguns economistas fizeram isso por nós. Eles randomizaram as aulas de forma que alguns alunos foram designados para ter aulas presenciais, outros para ter apenas aulas online e um terceiro grupo para ter um formato misto de aulas online e presenciais. Eles coletaram dados em um exame padrão no final do semestre.

Aqui está como os dados se parecem:

In [1]:
import pandas as pd
import numpy as np

data = pd.read_csv("./data/online_classroom.csv")
print(data.shape)
data.head()

(323, 10)


gender  asian  black  hawaiian  hispanic  unknown  white  format_ol  \
0       0    0.0    0.0       0.0       0.0      0.0    1.0          0   
1       1    0.0    0.0       0.0       0.0      0.0    1.0          0   
2       1    0.0    0.0       0.0       0.0      0.0    1.0          0   
3       1    0.0    0.0       0.0       0.0      0.0    1.0          0   
4       1    0.0    0.0       0.0       0.0      0.0    1.0          1   

   format_blended  falsexam  
0             0.0  63.29997  
1             0.0  79.96000  
2             1.0  83.37000  
3             1.0  90.01994  
4             0.0  83.30000

Podemos ver que temos 323 amostras. Não é exatamente uma grande quantidade de dados, mas é algo com o qual podemos trabalhar. Para estimar o efeito causal, podemos simplesmente calcular a pontuação média para cada um dos grupos de tratamento.

In [4]:
(data
 .assign(class_format = np.select(
     [data["format_ol"].astype(bool), data["format_blended"].astype(bool)],
     ["online", "blended"],
     default="face_to_face"
 ))
 .groupby(["class_format"])
 .mean())

gender     asian     black  hawaiian  hispanic   unknown  \
class_format                                                               
blended       0.550459  0.217949  0.102564  0.025641  0.012821  0.012821   
face_to_face  0.633333  0.202020  0.070707  0.000000  0.010101  0.000000   
online        0.542553  0.228571  0.028571  0.014286  0.028571  0.000000   

                 white  format_ol  format_blended   falsexam  
class_format                                                  
blended       0.628205        0.0             1.0  77.093731  
face_to_face  0.717172        0.0             0.0  78.547485  
online        0.700000        1.0             0.0  73.635263

Sim, é simples assim. Podemos ver que as aulas presenciais têm uma pontuação média de 78,54, enquanto os cursos online têm uma pontuação média de 73,63. Não são boas notícias para os defensores do aprendizado online. O $ATE$ para uma aula online é, portanto, -4,91. Isso significa que **as aulas online fazem com que os alunos tenham um desempenho cerca de 5 pontos mais baixo**, em média. É isso. Você não precisa se preocupar que os cursos online possam ter alunos mais pobres que não podem pagar por aulas presenciais ou, por essa razão, não precisa se preocupar que os alunos dos diferentes tratamentos sejam diferentes de alguma forma além do tratamento que receberam. Por design, o experimento aleatório é feito para eliminar essas diferenças.

Por essa razão, uma boa verificação de sanidade para ver se a randomização foi feita corretamente (ou se você está olhando para os dados corretos) é verificar se os tratados são iguais aos não tratados em variáveis pré-tratamento. Nossos dados têm informações sobre gênero e etnia para ver se eles são semelhantes entre os grupos. Podemos dizer que eles parecem bastante semelhantes para as variáveis `gênero`, `asiático`, `hispanico` e `branco`. No entanto, a variável `negro` parece um pouco diferente. Isso chama a atenção para o que acontece com um pequeno conjunto de dados. Mesmo sob randomização, pode ser que, por acaso, um grupo seja diferente de outro. Em amostras grandes, essa diferença tende a desaparecer.

## O Experimento Ideal

Experimentos randomizados ou Ensaios Clínicos Randomizados (ECR) são a maneira mais confiável de obter efeitos causais. É uma técnica direta e absurdamente convincente. É tão poderosa que a maioria dos países a tem como requisito para demonstrar a eficácia de novos medicamentos. Para fazer uma analogia terrível, você pode pensar no ECR como Aang, de Avatar: A Lenda de Aang, enquanto outras técnicas são mais como Sokka. Sokka é legal e pode fazer alguns truques interessantes aqui e ali, mas Aang pode dobrar os quatro elementos e se conectar com o mundo espiritual. Pense dessa forma, se pudéssemos, ECR seria tudo o que faríamos para descobrir a causalidade. Um ECR bem projetado é o sonho de qualquer cientista.

![img](./data/img/rct/science_dream.png)

Infelizmente, eles tendem a ser muito caros ou simplesmente antiéticos. Às vezes, simplesmente não podemos controlar o mecanismo de atribuição. Imagine-se como um médico tentando estimar o efeito do tabagismo durante a gravidez no peso do bebê ao nascer. Você não pode simplesmente forçar uma parcela aleatória de mães a fumar durante a gravidez. Ou digamos que você trabalhe para um grande banco e precise estimar o impacto do limite de crédito na rotatividade de clientes. Seria muito caro dar linhas de crédito aleatórias aos seus clientes. Ou que você queira entender o impacto do aumento do salário mínimo no desemprego. Você não pode simplesmente designar países para terem um ou outro salário mínimo. Você entendeu a ideia.

Veremos mais tarde como diminuir o custo da randomização usando randomização condicional, mas não há nada que possamos fazer sobre experimentos antiéticos ou inviáveis. Ainda assim, sempre que lidamos com questões causais, vale a pena pensar no **experimento ideal**. Sempre se pergunte, se pudesse, **qual seria o experimento perfeito que você realizaria para descobrir esse efeito causal?** Isso tende a lançar alguma luz sobre a maneira como podemos descobrir o efeito causal mesmo sem o experimento ideal.


## O Mecanismo de Atribuição

Em um experimento randomizado, o mecanismo que atribui unidades a um tratamento ou outro é, bem, aleatório. Como veremos mais tarde, todas as técnicas de inferência causal de alguma forma tentarão identificar os mecanismos de atribuição dos tratamentos. Quando sabemos com certeza como esse mecanismo se comporta, a inferência causal será muito mais confiável, mesmo que o mecanismo de atribuição não seja aleatório.

Infelizmente, o mecanismo de atribuição não pode ser descoberto simplesmente olhando para os dados. Por exemplo, se você tem um conjunto de dados onde a educação superior correlaciona com a riqueza, você não pode saber com certeza qual causou qual apenas olhando para os dados. Você terá que usar seu conhecimento sobre como o mundo funciona para argumentar em favor de um mecanismo de atribuição plausível: é o caso de as escolas educarem as pessoas, tornando-as mais produtivas e levando-as a empregos com salários mais altos. Ou, se você é pessimista sobre a educação, pode dizer que as escolas não fazem nada para aumentar a produtividade, e isso é apenas uma correlação espúria porque apenas famílias ricas podem pagar para que um filho obtenha um diploma superior.

Em questões causais, geralmente podemos argumentar de ambas as formas: que X causa Y, ou que é uma terceira variável Z que causa tanto X quanto Y, e, portanto, a correlação entre X e Y é apenas espúria. Por essa razão, conhecer o mecanismo de atribuição leva a uma resposta causal muito mais convincente. Isso também é o que torna a inferência causal tão empolgante. Enquanto a aprendizagem de máquina (ML) aplicada geralmente é apenas pressionar alguns botões na ordem correta, a inferência causal aplicada exige que você pense seriamente sobre o mecanismo que gera esses dados.


## Principais Ideias

Examinamos como os experimentos randomizados são a maneira mais simples e eficaz de descobrir o impacto causal. Isso é feito tornando os grupos de tratamento e controle comparáveis. Infelizmente, não podemos realizar experimentos randomizados o tempo todo, mas ainda é útil pensar sobre qual seria o experimento ideal que faríamos se pudéssemos.

Alguém familiarizado com estatísticas pode estar protestando agora que eu não olhei para a variância da minha estimativa de efeito causal. Como posso saber que uma diminuição de 4,91 pontos não é devido ao acaso? Em outras palavras, como posso saber se a diferença é estatisticamente significativa? E eles estariam certos. Não se preocupe. Pretendo revisar alguns conceitos estatísticos em seguida.


## Referências

Gosto de pensar nesta série inteira como uma homenagem a Joshua Angrist, Alberto Abadie e Christopher Walters por sua incrível aula de Econometria. A maioria das ideias aqui foram tiradas de suas aulas na Associação Econômica Americana. Assisti-las é o que está me mantendo são durante este difícil ano de 2020.
* [Cross-Section Econometrics](https://www.aeaweb.org/conference/cont-ed/2017-webcasts)
* [Mastering Mostly Harmless Econometrics](https://www.aeaweb.org/conference/cont-ed/2020-webcasts)

Também gostaria de referenciar os livros incríveis de Angrist. Eles me mostraram que Econometria, ou 'Métricas, como eles chamam, não é apenas extremamente útil, mas também profundamente divertida.

* [Mostly Harmless Econometrics](https://www.mostlyharmlesseconometrics.com/)
* [Mastering 'Metrics](https://www.masteringmetrics.com/)

Finalmente, gostaria de referenciar o livro de Miguel Hernan e Jamie Robins. Tem sido meu fiel companheiro nas questões mais espinhosas de inferência causal que tive que responder.

* [Causal Inference Book](https://www.hsph.harvard.edu/miguel-hernan/causal-inference-book/)

Os dados utilizados aqui são de um estudo de Alpert, William T., Kenneth A. Couch, and Oskar R. Harmon. 2016. ["A Randomized Assessment of Online Learning"](https://www.aeaweb.org/articles?id=10.1257/aer.p20161057). American Economic Review, 106 (5): 378-82.

![img](./data/img/poetry.png)

## Contribua

"Inferência Causal para os Corajosos e Verdadeiros" é um material de código aberto sobre inferência causal, a estatística da ciência. Seu objetivo é ser acessível monetariamente e intelectualmente. Ele utiliza apenas software gratuito baseado em Python.
Se você encontrou valor neste livro e deseja apoiá-lo, por favor, vá para o [Patreon](https://www.patreon.com/causal_inference_for_the_brave_and_true). Se você não estiver pronto para contribuir financeiramente, também pode ajudar corrigindo erros, sugerindo edições ou dando feedback sobre trechos que não compreendeu. Acesse o repositório do livro e [abra uma issue](https://github.com/matheusfacure/python-causality-handbook/issues). Por fim, se você gostou deste conteúdo, compartilhe com outras pessoas que possam achar útil e dê uma [estrela no GitHub](https://github.com/matheusfacure/python-causality-handbook/stargazers).

---

<div align="center">
<a href="01-Introduction-To-Causality.ipynb"><-- Anterior</a>  
<a href="00-Summary.ipynb">| Sumário |</a>  
<a href="03-Stats-Review-The-Most-Dangerous-Equation.ipynb">Próximo --></a>  
</div>